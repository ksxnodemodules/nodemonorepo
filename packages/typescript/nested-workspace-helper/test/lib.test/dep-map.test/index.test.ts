import {getDependencyMap} from '../../../index'
import createSetupTeardown from '../../.lib/setup-teardown'

const {apply} = createSetupTeardown('valid.yaml')
const root = 'root'

describe('getDependencyMap', () => {
  it('matches snapshot', apply(async () => {
    expect(await getDependencyMap(root)).toMatchSnapshot()
  }))
})
