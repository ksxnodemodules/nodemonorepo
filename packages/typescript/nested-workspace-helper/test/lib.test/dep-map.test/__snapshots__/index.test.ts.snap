// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDependencyMap matches snapshot 1`] = `
Object {
  "root": Object {
    "dependant": Object {
      "manifestContent": Object {
        "devDependencies": Object {
          "pkg00": "^0.1.0",
        },
        "name": "root",
        "private": true,
        "version": "0.0.0",
        "workspaces": Array [
          "wrkspc0dept/*",
          "wrkspc1dept/*/*",
          "wrkspc2dept/*/*/*",
        ],
      },
      "manifestFile": "root/package.json",
      "path": "root",
    },
    "list": Array [
      Object {
        "info": Object {
          "manifestContent": Object {
            "name": "pkg00",
            "version": "0.1.2",
          },
          "manifestFile": "root/wrkspc0dept/pkg00/package.json",
          "path": "root/wrkspc0dept/pkg00",
        },
        "name": "pkg00",
        "requirement": "^0.1.0",
        "type": "dev",
        "version": "0.1.2",
      },
    ],
  },
  "root/wrkspc0dept/pkg00": Object {
    "dependant": Object {
      "manifestContent": Object {
        "name": "pkg00",
        "version": "0.1.2",
      },
      "manifestFile": "root/wrkspc0dept/pkg00/package.json",
      "path": "root/wrkspc0dept/pkg00",
    },
    "list": Array [],
  },
  "root/wrkspc0dept/pkg01": Object {
    "dependant": Object {
      "manifestContent": Object {
        "name": "pkg01",
        "private": false,
        "version": "1.2.3",
      },
      "manifestFile": "root/wrkspc0dept/pkg01/package.json",
      "path": "root/wrkspc0dept/pkg01",
    },
    "list": Array [],
  },
  "root/wrkspc1dept/sub10/pkg100": Object {
    "dependant": Object {
      "manifestContent": Object {
        "name": "pkg100",
        "version": "2.2.5",
      },
      "manifestFile": "root/wrkspc1dept/sub10/pkg100/package.json",
      "path": "root/wrkspc1dept/sub10/pkg100",
    },
    "list": Array [],
  },
  "root/wrkspc1dept/sub11/pkg110": Object {
    "dependant": Object {
      "manifestContent": Object {
        "dependencies": Object {
          "pkg01": "^1.0.0",
          "pkg100": "~2.2.1",
        },
        "name": "pkg110",
        "version": "22.3.5",
      },
      "manifestFile": "root/wrkspc1dept/sub11/pkg110/package.json",
      "path": "root/wrkspc1dept/sub11/pkg110",
    },
    "list": Array [
      Object {
        "info": Object {
          "manifestContent": Object {
            "name": "pkg01",
            "private": false,
            "version": "1.2.3",
          },
          "manifestFile": "root/wrkspc0dept/pkg01/package.json",
          "path": "root/wrkspc0dept/pkg01",
        },
        "name": "pkg01",
        "requirement": "^1.0.0",
        "type": "prod",
        "version": "1.2.3",
      },
      Object {
        "info": Object {
          "manifestContent": Object {
            "name": "pkg100",
            "version": "2.2.5",
          },
          "manifestFile": "root/wrkspc1dept/sub10/pkg100/package.json",
          "path": "root/wrkspc1dept/sub10/pkg100",
        },
        "name": "pkg100",
        "requirement": "~2.2.1",
        "type": "prod",
        "version": "2.2.5",
      },
    ],
  },
  "root/wrkspc1dept/sub11/pkg111prv": Object {
    "dependant": Object {
      "manifestContent": Object {
        "dependencies": Object {
          "pkg01": "*",
        },
        "name": "pkg111prv",
        "private": true,
        "version": "33.22.11",
      },
      "manifestFile": "root/wrkspc1dept/sub11/pkg111prv/package.json",
      "path": "root/wrkspc1dept/sub11/pkg111prv",
    },
    "list": Array [
      Object {
        "info": Object {
          "manifestContent": Object {
            "name": "pkg01",
            "private": false,
            "version": "1.2.3",
          },
          "manifestFile": "root/wrkspc0dept/pkg01/package.json",
          "path": "root/wrkspc0dept/pkg01",
        },
        "name": "pkg01",
        "requirement": "*",
        "type": "prod",
        "version": "1.2.3",
      },
    ],
  },
  "root/wrkspc2dept/sub20/sub200/pkg2000": Object {
    "dependant": Object {
      "manifestContent": Object {
        "devDependencies": Object {
          "foreign-foo": "^1.2.3",
          "pkg111prv": "33.22.11",
        },
        "name": "pkg2000",
        "version": "123.456.789",
      },
      "manifestFile": "root/wrkspc2dept/sub20/sub200/pkg2000/package.json",
      "path": "root/wrkspc2dept/sub20/sub200/pkg2000",
    },
    "list": Array [
      Object {
        "info": Object {
          "manifestContent": Object {
            "dependencies": Object {
              "pkg01": "*",
            },
            "name": "pkg111prv",
            "private": true,
            "version": "33.22.11",
          },
          "manifestFile": "root/wrkspc1dept/sub11/pkg111prv/package.json",
          "path": "root/wrkspc1dept/sub11/pkg111prv",
        },
        "name": "pkg111prv",
        "requirement": "33.22.11",
        "type": "dev",
        "version": "33.22.11",
      },
    ],
  },
}
`;
